### PR TITLE
add nolzma and nozstd config option for boost.iostreams

### PR DESCRIPTION
**Pull request by xiongyang**
add nolzma and nozstd config option
default is true  which will disable using zstd and xz
 if set as false it will add_deps xz/zstd then complie boost with the deps

### CODE REVIEW

File: packages/b/boost/xmake.lua [hunk 1]
```diff
@@ -78,6 +78,8 @@ package("boost")
     for _, libname in ipairs(libnames) do
         add_configs(libname,    { description = "Enable " .. libname .. " library.", default = (libname == "filesystem"), type = "boolean"})
     end
+    add_configs("nozstd", {description = "disable zstd for iostreams", default = true, type = "boolean"})
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:81]: please rename `nozstd` -> `zstd`, set the default value: false

File: packages/b/boost/xmake.lua [hunk 2]
```diff
@@ -78,6 +78,8 @@ package("boost")
...
+    add_configs("nolzma", {description = "disable lzma for iostreams", default = true, type = "boolean"})
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:82]: and here

Comment (xiongyang) [reply packages/b/boost/xmake.lua]: ok. all neme changed. also  update all the if statement

File: packages/b/boost/xmake.lua [hunk 3]
```diff
@@ -242,9 +252,21 @@ package("boost")
             end
         end
 
+        local function config_deppath(file,depName,ruleName)
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:255]: add spaces ` , `, and use lower variable name

File: packages/b/boost/xmake.lua [hunk 4]
```diff
@@ -242,9 +252,21 @@ package("boost")
...
+                local dep = package:dep(depName)
+                local info = dep:fetch({external = false})
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:257]: info maybe nil, we should add `if info then`

File: packages/b/boost/xmake.lua [hunk 5]
```diff
@@ -242,9 +252,21 @@ package("boost")
...
+                local usStr = format("\nusing %s : : <include>\"%s\" <search>\"%s\" ;",ruleName, info.includedirs[1], info.linkdirs[1])
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:258]: use lowercase, and includedirs may be string instead of table.

File: packages/b/boost/xmake.lua [hunk 6]
```diff
@@ -406,4 +434,30 @@ package("boost")
...
             ]]}, {configs = {languages = "c++14"}}))
         end
+
+        if package:config("iostreams") then
+            if package:config("zstd") then
+                assert(package:check_cxxsnippets({test = [[
+                #include <boost/iostreams/filter/zstd.hpp>
+                #include <boost/iostreams/filtering_stream.hpp>
+                int main(int argc, char* argv[]) {
+                boost::iostreams::filtering_ostream out;
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:444]: incorrect indent

File: packages/b/boost/xmake.lua [hunk 7]
```diff
@@ -406,4 +434,30 @@ package("boost")
                 }
             ]]}, {configs = {languages = "c++14"}}))
         end
+
+        if package:config("iostreams") then
+            if package:config("zstd") then
+                assert(package:check_cxxsnippets({test = [[
+                #include <boost/iostreams/filter/zstd.hpp>
+                #include <boost/iostreams/filtering_stream.hpp>
+                int main(int argc, char* argv[]) {
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:443]: please use `void test()` instead of `main`

File: packages/b/boost/xmake.lua [hunk 8]
```diff
@@ -270,6 +290,14 @@ package("boost")
             "runtime-debugging=" .. (package:is_debug() and "on" or "off")
         }
 
+        if not package:config("lzma") then
+            table.insert(argv,"-sNO_LZMA=1")
+        end
+        if not package:config("zstd") then
+            table.insert(argv,"-sNO_ZSTD=1")
+        end
+
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:299]: remove an empty line

File: packages/b/boost/xmake.lua [hunk 9]
```diff
@@ -270,6 +290,14 @@ package("boost")
             "runtime-debugging=" .. (package:is_debug() and "on" or "off")
         }
 
+        if not package:config("lzma") then
+            table.insert(argv,"-sNO_LZMA=1")
```
Comment (waruqi) [inline packages/b/boost/xmake.lua:294]: add spaces, ` , `

File: packages/b/boost/xmake.lua (hunk 5 above)
Comment (xiongyang) [reply packages/b/boost/xmake.lua]: fix the style and format is done.
includedirs is table with only one element for xz and zstd

**Pull request merged**